datasets/za/wanted/crawler.py

```diff
@@ -0,0 +1,62 @@
+from zavod import Context
+from lxml import html
+from urllib.parse import urlparse, parse_qs
+import re
+
+from zavod import helpers as h
+
+REGEX_PATTERN = re.compile("(.+)\((.+)\)(.+)")
+
+def crawl_person(context: Context, cell: html.HtmlElement):
+    source_url = cell.xpath(".//a/@href")[0]
+    match = REGEX_PATTERN.match(cell.text_content())
+
+    if not match:
+        context.log.warning("Regex did not match data for person %s" % source_url)
+        return
+
+    name, crime, status = map(str.strip, match.groups())
+    last_name, first_name = map(str.strip, name.split(",", maxsplit=1))
+
+    # either first or last name is considered a bare minimum to emit a person entity
+    unknown_spellings = ["Unknown", "Uknown"]
+    if first_name in unknown_spellings and last_name in unknown_spellings:
+        return
+
+    if "," in first_name or "," in last_name:
+        context.log.warning(
+            "Possible problem parsing person name (extra comma)",
+            first_name=first_name,
+            last_name=last_name,
+            source_url=source_url,
+        )
+
+    person = context.make("Person")
+    
+    # each wanted person has a dedicated details page 
+    # which appears to be a unique identifier
+    id = parse_qs(urlparse(source_url).query)["bid"][0]
+    person.id = context.make_slug(id)
+
+    h.apply_name(
+        person,
+        first_name=first_name,
+        last_name=last_name,
+    )
+
+    person.add("sourceUrl", source_url)
+    person.add("notes", f"{status} - {crime}")
+
+    person.add("topics", "crime")
+    person.add("country", "za")
```

Comment:
> yeah I think if there's either a first or a last name, it's fair to emit them. I think to make this a bit less risky, we can add `za` to the `country` property.

@jbothma I've added `za` to the `country` property but I'm wondering what are the semantics in this case? I've looked at the [data dictionary](https://www.opensanctions.org/reference/#schema.Person) but still couldn't understand. Or this is simply to add information to the [deduplication](https://www.opensanctions.org/articles/2021-11-11-deduplication/) process?

Reply:
yeah my reasoning here is both when we're doing deduplication, and if people are doing matching against their customers and someone from this list with a partial name comes up, they can more easily determine if it's a true match.